src/lammpsio/snapshot.py

```diff
@@ -91,22 +91,77 @@ def from_hoomd_gsd(cls, frame):
         snap.charge = frame.particles.charge
         snap.mass = frame.particles.mass
 
-        snap.molecule = frame.particles.body + 1
+        if frame.particles.body is not None:
+            snap.molecule = frame.particles.body + 1
         if numpy.any(snap.molecule < 0):
             warnings.warn("Some molecule IDs are negative, remapping needed.")
 
-        if (
-            frame.bonds.N > 0
-            or frame.angles.N > 0
-            or frame.dihedrals.N > 0
-            or frame.impropers.N > 0
-        ):
-            warnings.warn("Conversion of topology from gsd is not supported")
-        type_map = {typeid + 1: i for typeid, i in enumerate(frame.particles.types)}
+        if frame.bonds.N != 0:
+            snap.bonds = Bonds(N=frame.bonds.N)
+            snap.bonds.id = frame.bonds.typeid
+            snap.bonds.members = frame.bonds.group + 1
+
+        if frame.angles.N != 0:
+            snap.angles = Angles(N=frame.angles.N)
+            snap.angles.id = frame.angles.typeid
+            snap.angles.members = frame.angles.group + 1
+
+        if frame.dihedrals.N != 0:
+            snap.dihedrals = Dihedrals(N=frame.dihedrals.N)
+            snap.dihedrals.id = frame.dihedrals.typeid
+            snap.dihedrals.members = frame.dihedrals.group + 1
+
+        if frame.impropers.N != 0:
+            snap.impropers = Impropers(N=frame.impropers.N)
+            snap.impropers.id = frame.impropers.typeid
+            snap.impropers.members = frame.impropers.group + 1
+
+        # set particle label
+        label_map_particle = None
+        if frame.particles.types is not None:
+            label_map_particle = {
+                typeid + 1: i for typeid, i in enumerate(frame.particles.types)
+            }
+        snap.label = LabelMap(map=label_map_particle)
```

Comment:
This would go best above (near where the old `type_map` was deleted), before dealing with the bonded interactions.